gtest: print backtraces on sigill

rockwotj opened the pull request:
https://redpandadata.slack.com/archives/C07HD9U0EHL/p1763091066630739?thread_ts=1763061976.219569&cid=C07HD9U0EHL

## Backports Required

- [x] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.3.x
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes

* none

rockwotj committed:
gtest: give stacktraces on sigill

Use absl's signal handler in gtest in order to give proper backtraces.
We don't use crash_tracker because it's undesirable to bring in a bunch
of dependencies for such a low level thing. If someone wanted to extract
out just the backtrace stuff for tests from crash_tracker, I'd accept
it, but for now this will do.

WillemKauf reviewed (dismissed):
:shipit:

nvartolomei commented:
Do we want/need to handle sigv there too? What's the boost test behavior? We still have quite a few of those.

pgellert commented on src/v/test_utils/gtest_main.cc:
```diff
@@ -8,14 +8,35 @@
     testing::InitGoogleTest(&argc, argv);
     GTEST_FLAG_SET(death_test_style, "threadsafe");
+    // Fall back signal handlers if our crash tracker isn't registered
+    absl::InstallFailureSignalHandler({});
```
In debug builds, how does this interact with sanitizers? https://github.com/redpanda-data/redpanda/blob/eb46f30376ceaaff3419b4f14cc3edd330ba3b2f/bazel/test.bzl#L78-L81

I think it's probably best to use `absl::InstallFailureSignalHandler(absl::FailureSignalHandlerOptions{.call_previous_handler = true})` here to wrap any existing signal handlers.

rockwotj replied:
The sanitizers still work just fine from my testing. We don't need to do anything there. It's only triggering on SIGILL in debug mode for me.

rockwotj commented:
> Do we want/need to handle sigv there too?

SIGSEGV is already unblocked, it's just SIGILL

>  What's the boost test behavior? We still have quite a few of those.

We can and should do another PR for that

rockwotj commented:
Hrm seems this signal handler stuff is breaking the DEATH_TESTs 😢

rockwotj committed:
bazel: enable absl integration for googletest

Silly enough we have to set this define as a flag to get googletest to
depend on absl. This enables flags like `install_failure_signal_handler`
to work correctly.

rockwotj committed:
protobuf/test: use public API to get test flag value

rockwotj commented:
/ci-repeat
